packages/amplify_api/lib/graphql/graphql_helpers.dart

```diff
@@ -0,0 +1,131 @@
+
+import 'package:amplify_api_plugin_interface/amplify_api_plugin_interface.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+import '../amplify_api.dart';
+
+class ModelQueriesFactory extends ModelQueriesInterface {
+  @override
+  GraphQLRequest<T> get<T extends Model>(ModelType<T> modelType, String id) {
+    Map<String, String> variableInput = {"id": "ID"};
+
+    return GraphQLRequestFactory().buildQuery<T>(
+        modelType: modelType,
+        variableInput: variableInput,
+        id: id,
+        requestType: GraphQLRequestType.query,
+        requestOperation: GraphQLRequestOperation.get);
+  }
+}
+
+class GraphQLRequestFactory extends GraphQLRequestFactoryInterface {
+  String _getModelType(ModelFieldTypeEnum? val) {
+    switch (val) {
+      case ModelFieldTypeEnum.string:
+        return "String";
+      case ModelFieldTypeEnum.int:
+        return "int";
+      case ModelFieldTypeEnum.model:
+        return "ID";
+      default:
+        return "Error: UnknownType!";
+    }
+  }
+
+  List<String> _getFieldsFromModelType(Map<String, ModelField?>? fieldsMap) {
+    List<String> fieldsList = [];
+    if (fieldsMap != null) {
+      fieldsMap.forEach((key, value) {
+        if (value?.association == null) fieldsList.add(key);
+      });
+    }
+    return fieldsList;
+  }
+
+  ModelSchema _getSchema(ModelType modelType) {
+    ModelProviderInterface? provider = AmplifyAPI.instance.getModelProvider();
+
+    if (provider == null) {
+      throw ApiException('No modelProvider found',
+          recoverySuggestion:
+              'Pass in a modelProvider instance while instantiating APIPlugin');
+    }
+
+    ModelSchema schema = provider.modelSchemas.firstWhere(
+        (elem) => provider.getModelTypeByModelName(elem.name) == modelType);
+
+    if (schema == null) {
+      throw ApiException('No schema found for the ModelType provided',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a vaild ModelType');
+    }
+
+    return schema;
+  }
+
+  @override
+  GraphQLRequest<T> buildQuery<T extends Model>(
+      {required ModelType modelType,
+      required Map<String, String>? variableInput,
+      required String? id,
+      required GraphQLRequestType requestType,
+      required GraphQLRequestOperation requestOperation}) {
+    ModelSchema schema = _getSchema(modelType);
+
+    String? modelName = requestOperation == GraphQLRequestOperation.list
+        ? schema.pluralName
+        : schema.name;
+    String name = modelName ?? 'No model name found';
+    String fields = _getFieldsFromModelType(schema.fields).join(' ');
+
+    String reqTypeStr = describeEnum(requestType.toString());
+    String reqOperationStr = describeEnum(requestOperation.toString());
+
+    var varInputUpperStr = '';
+    var varInputLowerStr = '';
+
+    if (variableInput!.isNotEmpty) {
```

Comment:
What happens here if `variableInput` is null? Won't it be an error? Can `variableInput` just be non-nullable?

Reply:
Updated the conditional so the logic makes more sense. `variableInput` is an optional parameter so we need to make sure its exists and has value.